src/main/java/com/google/opensesame/user/UserData.java

```diff
@@ -110,30 +113,24 @@ public String getUserID() {
     return userID;
   }
 
-  /** @return a list of ids of projects a user is involved in. */
-  public ArrayList<String> getProjectIDs() {
-    return projectIds;
-  }
-
-  /** @return a list of ids of mentees a user is currently mentoring. */
-  public ArrayList<String> getMenteeIDs() {
-    return menteeIds;
+  /** @return a list of of projects a user is involved in. */
+  public ArrayList<ProjectData> getProjects() {
+    return projects;
   }
 
   /**
-   * Add a project to the list of projects associated with this instance of UserData. Remember to
-   * store this change in datastore with a UserEntity.
-   */
-  public void addProject(String projectId) {
-    projectIds.add(projectId);
-    isMentor = true;
-  }
-
-  /**
-   * Add a project to the list of projects associated with this instance of UserData. Remember to
-   * store this change in datastore with a UserEntity.
+   * Add the {@code ProjectData} associated with the projects in {@code projectIds} to {@code
+   * this.projects}.
+   *
+   * @param projectIds
    */
-  public void addMentee(String menteeId) {
-    projectIds.add(menteeId);
+  private void generateProjects(ArrayList<String> projectIds) throws IOException {
```

Comment:
How about naming this `loadProjects` instead? The name `generateProjects` makes me thing that somehow there is logic in this function that creates the projects without using pre-stored data.